### DISCUSSION TITLE
Find whether player is looking at certain entity or block

### DISCUSSION THREAD

**Discussion by LooFifteen** [Minestom/Minestom#796] [Help] [answered]
How would I check if a player is looking at a certain entity or block, and differentiate between the two.

**Comment by emortaldev** [comment, marked as answer]
Raycasts! I believe Minestom has a few built in raycasting stuff from block collisions, however if that isn't enough you can check out [Rayfast](https://github.com/EmortalMC/Rayfast)

**Reply by LooFifteen** [reply]
Any examples from either solution?

**Reply by emortaldev** [reply]
[Here's an example for rayfast](https://github.com/EmortalMC/LazerTag/blob/29a90a26be18f26c98cee77d592d1f806f412b4c/src/main/kotlin/dev/emortal/lazertag/raycast/RaycastUtil.kt#L120) (from LazerTag)

**Reply by emortaldev** [reply]
(just realised it could be optimised lol)
